From 76e85fbedf8d8d361ab9fae49ac318f5b7533783 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Wed, 29 May 2024 11:03:13 +0200
Subject: [PATCH] dt-bindings: iio: adc: stm32-adc: add support for stm32mp21

Add stm32mp21 compatible for STM32MP21 SoC. It has 2 ADCs: ADC1 and
ADC2. It also has a new trigger list for each ADCs and also differ
by their internal signal from ADCs in STM32MP25 and it has no 'vddgpu'
internal voltage channel.

Change-Id: Ia34296b27dfe0dcdeb91a079241389c15c41f955
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/383347
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 .../bindings/iio/adc/st,stm32-adc.yaml        | 60 ++++++++++++++++++-
 1 file changed, 59 insertions(+), 1 deletion(-)

--- a/Documentation/devicetree/bindings/iio/adc/st,stm32-adc.yaml
+++ b/Documentation/devicetree/bindings/iio/adc/st,stm32-adc.yaml
@@ -28,6 +28,7 @@ properties:
       - st,stm32h7-adc-core
       - st,stm32mp1-adc-core
       - st,stm32mp13-adc-core
+      - st,stm32mp21-adc-core
       - st,stm32mp25-adc-core
 
   reg:
@@ -221,6 +222,31 @@ allOf:
       properties:
         compatible:
           contains:
+            const: st,stm32mp21-adc-core
+
+    then:
+      properties:
+        clocks:
+          maxItems: 1
+
+        clock-names:
+          const: adc
+
+        interrupts:
+          items:
+            - description: interrupt line for ADC1
+            - description: interrupt line for ADC2
+          minItems: 1
+
+        st,max-clk-rate-hz:
+          minimum: 700000
+          maximum: 70000000
+          default: 70000000
+
+  - if:
+      properties:
+        compatible:
+          contains:
             const: st,stm32mp25-adc-core
 
     then:
@@ -271,6 +297,7 @@ patternProperties:
           - st,stm32h7-adc
           - st,stm32mp1-adc
           - st,stm32mp13-adc
+          - st,stm32mp21-adc
           - st,stm32mp25-adc
 
       reg:
@@ -323,7 +350,7 @@ patternProperties:
             - 16 channels, numbered from 0 to 15 (for in0..in15) on stm32f4
             - 19 channels, numbered from 0 to 18 (for in0..in18) on stm32mp13.
             - 20 channels, numbered from 0 to 19 (for in0..in19) on stm32h7,
-              stm32mp1 and stm32mp25.
+              stm32mp1, stm32mp21 and stm32mp25.
         $ref: /schemas/types.yaml#/definitions/uint32-array
         deprecated: true
 
@@ -518,6 +545,37 @@ patternProperties:
 
       - if:
           properties:
+            compatible:
+              contains:
+                const: st,stm32mp21-adc
+
+          then:
+            properties:
+              reg:
+                const: 0x0
+
+              interrupts:
+                const: 0
+
+              assigned-resolution-bits:
+                enum: [6, 8, 10, 12]
+                default: 12
+
+              st,adc-channels:
+                minItems: 1
+                maxItems: 20
+                items:
+                  minimum: 0
+                  maximum: 19
+
+              st,min-sample-time-nsecs:
+                minItems: 1
+                maxItems: 20
+                items:
+                  minimum: 40
+
+      - if:
+          properties:
             compatible:
               contains:
                 const: st,stm32mp25-adc
